pkg/api/validate/openshiftcluster_validatedynamic.go

```diff
@@ -163,12 +172,9 @@ func (dv *openShiftClusterDynamicValidator) validateVnetPermissions(ctx context.
 			"Microsoft.Network/virtualNetworks/subnets/write",
 		} {
 			ok, err := utilpermissions.CanDoAction(permissions, action)
-			if err != nil {
+			if !ok || err != nil {
 				return false, err
 			}
-			if !ok {
-				return false, nil
```

Comment:
`err == nil` here, so this block could be
```
if !ok {
    return false, err
}
```

And then it can just be combined with the above block using `||` between the conditions.